runtime-light/stdlib/string/pcre2-functions.h

```diff
@@ -0,0 +1,78 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2024 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <optional>
+#include <string_view>
+
+#include "runtime-common/core/allocator/script-allocator.h"
+#include "runtime-common/core/runtime-core.h"
+#include "runtime-common/core/std/containers.h"
+#include "runtime-common/stdlib/string/mbstring-functions.h"
+#include "runtime-light/stdlib/string/regex-include.h"
+
+namespace kphp::pcre2 {
+
+using group_names_t = kphp::stl::vector<const char*, kphp::memory::script_allocator>;
+
+class match_view {
+public:
+  match_view(std::string_view subject, PCRE2_SIZE* ovector, int32_t ret_code) noexcept
+      : m_subject_data{subject},
+        m_ovector_ptr{ovector},
+        m_num_groups{ret_code} {}
+
+  match_view(const match_view& other) noexcept = default;
+  match_view(match_view&& other) noexcept = default;
+  match_view& operator=(const match_view& other) noexcept = default;
+  match_view& operator=(match_view&& other) noexcept = default;
+
+  int32_t size() const noexcept {
+    return m_num_groups;
+  }
+
+  std::optional<std::string_view> get_group(size_t i) const noexcept;
+
+private:
+  std::string_view m_subject_data;
+  const PCRE2_SIZE* m_ovector_ptr;
+  int32_t m_num_groups;
+};
+
+class compiled_regex {
+public:
+  static const compiled_regex* compile(const string& regex) noexcept;
+
+  bool is_utf() const noexcept {
+    return compile_options & PCRE2_UTF;
+  }
+
+  bool validate(std::string_view subject) const noexcept {
+    // UTF-8 validation
+    return !is_utf() || mb_UTF8_check(subject.data());
```

Comment:
Something is very wrong here... Why is `PHP` semantics used in `pcre2`-related API?

Reply:
reverted